src/IoC/InstancePool.cs

```diff
@@ -0,0 +1,50 @@
+﻿namespace IoC;
+
+public sealed class InstancePool
+{
+    //This is where the instances will be stored.
+    private static Dictionary<string, object>? _implementationStore;
+    //Make the contructor unaccessible.
+    InstancePool()
+    {
+        _implementationStore = new Dictionary<string, object>();
+    }
+
+
+    //This is the only way to access to the methods of the class.
+    private static InstancePool? _instance = null;
+    public static InstancePool Instance
+    {
+        get
+        {
+            if (_instance is null)
```

Comment:
I think you can convert this into a ternary operation in a single line

Reply:
Got it. Will update.